lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -481,25 +481,11 @@ def active_record_version_is?(version_requirement)
           requirement.satisfied_by?(version)
         end
 
-        def primary_key_query(id = nil)
-          id ||= send(self.class.primary_key)
-          ["#{quoted_table_name}.#{self.class.primary_key} != ?", id]
-        end
-
-        def composite_primary_key_query(id = nil)
-          keys = self.class.primary_key
-          if id
-            values = id
-          else
-            values = keys.map { |key| send(key) }
-          end
-
-          conditions = keys.map { |key| "#{quoted_table_name}.#{key} != ?" }.join(' AND ')
-          [conditions, *values]
-        end
+        def primary_key_condition(id = nil)
+          primary_keys = Array.wrap(self.class.primary_key)
 
-        def composite_primary_key?
-          self.class.respond_to?(:composite_primary_key?) && self.class.composite_primary_key?
+          id = id ? Array.wrap(id) : Array.wrap(primary_keys.map { |k| send(k) })
```

Comment:
Thanks for the update. Did you try `slice(*primary_keys)`? With Active Record objects it returns a hash with keys that you provide and the values as they are on the object.

https://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-i-slice

This line could then read: 

```
id ? primary_keys.zip(Array.wrap(id)) : slice(*primary_keys)
```

Reply:
Thanks for the advice. That works indeed. I didn't know that and I thought the default ruby slice method. No wonder how it didn't work for me : D
I changed it now and it seems to work.

I am still refactoring the tests to include scope, but not sure if I'm just complicating it too much  in order to reuse _test/shared_list.rb_ I'll try to clean it up a bit  before commiting it.

Reply:
Thanks @divagueame. Yea I didn't write these tests and they've kind of grown into a mess over time. I think it's ok to just repeat some scope change tests in your seperate file for sanity. Single ID primary keys are used directly in the tests which make things brittle. I suppose that test suite could be configured to use a single id and a composite in two seperate runs via some kind of configuration.

Reply:
I refactored Shared::List so that the Mixin used can be overwritten. I had to add a helper method to be able to create records with composite primary key. Not sure if it's better than having a separate file with some duplicated tests.  Let me know what you think.